sherpa/astro/data.py

```diff
@@ -2721,6 +2723,20 @@ def _channel_to_wavelength(self, val, group=True, response_id=None):
     """The identifier for the background component when not set."""
 
     def _fix_background_id(self, id):
+        """Identify the backround identifier.
+
+        Parameters
+        ----------
+        id : int, str, or None
+            The background identifier, or None. If None then the
+            default_background_id value will be used.
+
+        Returns
+        -------
+        bkg_id : int
```

Comment:
What is the return value (bkg_id) If id is passed in as a string?

Reply:
Yeah. This should be "`int or str`". The existing documentation is a bit unclear on this - I had thought that it could only be an int for historical reasons but it turns out not to be so (technically it could be "any type that we can use as a key for a dict" but I think it makes sense to force some structure on this).

So I should review the docs to look at this as part of this.

Reply:
I've made some changes (in general it looks like we do say "int or str" in most places).